packages/rooks/src/hooks/useHighlight.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * useHighlight
+ * @description search for a keyword in the DOM and highlight them
+ * @see {@link https://rooks.vercel.app/docs/useHighlight}
+ */
+import { useEffect, useRef } from "react";
+
+function useHighlight({
+  isReady = false,
+  keyword = "",
+}: {
+  isReady: boolean;
+  keyword: string;
+}) {
+  const ref = useRef<HTMLElement>(null);
+
+  useEffect(() => {
+    if (keyword && isReady && ref.current) {
+      const contentElement = ref.current;
+      if (contentElement) {
+        const text = contentElement.innerHTML;
+        const regex = new RegExp(keyword, "g");
+        const newText = text.replace(
+          regex,
+          `<mark class="highlighted">${keyword}</mark>`
```

Comment:
I was wondering if we can somehow get the DOM from another option and default the DOM to `<mark class="highlighted">{}</mark>`

```tsx
useHighlight(keyword, {
    render:(match) => `<mark class="highlighted">${keyword}</mark>`
})
```

Reply:
I wonder if I have understood your point correctly. I have added a render argument for cases where we want to include optional markup for the keywords.